src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -337,8 +337,75 @@ def _check_registry_health(cmd, registry_name, ignore_errors):
+    import socket
+    from msrestazure.tools import parse_resource_id, is_valid_resource_id, resource_id
+    from azure.cli.core.profiles import ResourceType
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+    if registry_name is None:
+        logger.warning("Registry name must be provided to verify DNS routings of its private endpoints")
+        return
+
+    registry = None
+
+    # retrieve registry
+    try:
+        registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    except CLIError:
+        logger.warning('Registry resource of "%s" must be accessible to verify DNS routings of its private endpoints.',
+                       registry_name)
+        return
+
+    if not registry.private_endpoint_connections:
+        logger.warning('Registry "%s" doesn\'t have private endpoints to verify DNS routings.', registry_name)
+        return
+
+    if not is_valid_resource_id(vnet_of_private_endpoint):
+        res = parse_resource_id(registry.id)
+        vnet_of_private_endpoint = resource_id(name=vnet_of_private_endpoint, resource_group=res['resource_group'],
+                                               namespace='Microsoft.Network', type='virtualNetworks',
+                                               subscription=res['subscription'])
+
+    # retrieve FQDNs for registry and its data endpoint
+    pe_ids = [e.private_endpoint and e.private_endpoint.id for e in registry.private_endpoint_connections]
+    dns_mappings = {}
+    for pe_id in pe_ids:
+        res = parse_resource_id(pe_id)
+        network_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_NETWORK,
+                                                 subscription_id=res['subscription'])
+        pe = network_client.private_endpoints.get(res['resource_group'], res['name'])
+        if bool(re.match(vnet_of_private_endpoint, pe.subnet.id, re.I)):
+            nic_id = pe.network_interfaces[0].id
+            nic_res = parse_resource_id(nic_id)
+            nic = network_client.network_interfaces.get(nic_res['resource_group'], nic_res['name'])
+            for dns_config in nic.ip_configurations:
+                if dns_config.private_link_connection_properties.fqdns[0] in dns_mappings:
+                    err = ('Registry "%s" has more than one private endpoint in the vnet of "%s".'
+                           ' DNS routing will be unreliable')
+                    logger.warning(err, registry_name, vnet_of_private_endpoint)
+                dns_mappings[dns_config.private_link_connection_properties.fqdns[0]] = dns_config.private_ip_address
+
+    dns_ok = True
+    if not dns_mappings:
+        logger.warning('Registry "%s" doesn\'t have private endpoints in the vnet of "%s".',
+                       registry_name, vnet_of_private_endpoint)
+        return
+
+    for fqdn in dns_mappings:
+        result = socket.gethostbyname(fqdn)
+        if result not in dns_mappings[fqdn]:
```

Comment:
Trying to understand why we're using `not in` here. `result` will contain IP V4 address string right? `dns_mappings[fqdn]` is also string not a list. Any reason for not doing `if result != dns_mappings[fqdn]:`? 

Also, if an invalid/non-existing FQDN is passed to `socket.gethostbyname`, it leads to an error. Should the error be caught?

![image](https://user-images.githubusercontent.com/20955088/115629725-cacbc980-a2b7-11eb-9bf0-1ace9214eead.png)

Reply:
Will address both comments